src/service/vsync-impl.js

```diff
@@ -309,6 +334,14 @@ export class Vsync {
         tasks[i].mutate(states[i]);
       }
     }
+    // Swap last arrays into double buffer.
+    this.nextTasks_ = tasks;
```

Comment:
`this.tasks_` can be modified during the loop above, so unless you want want to use a weird splice magic, you need 2 arrays.